autoconf/m4/nl_enable_optimization.m4

```diff
@@ -81,12 +82,13 @@ AC_DEFUN([NL_ENABLE_OPTIMIZATION],
             
                     fi
                 ])
-
-            if test "${nl_cv_build_optimized}" = "no"; then
-                CFLAGS="`echo ${CFLAGS} | sed -e 's,-O[[[:alnum:]]]*,-O0,g'`"
-                CXXFLAGS="`echo ${CXXFLAGS} | sed -e 's,-O[[[:alnum:]]]*,-O0,g'`"
-                OBJCFLAGS="`echo ${OBJCFLAGS} | sed -e 's,-O[[[:alnum:]]]*,-O0,g'`"
-                OBJCXXFLAGS="`echo ${OBJCXXFLAGS} | sed -e 's,-O[[[:alnum:]]]*,-O0,g'`"
-            fi
     ])
+
+    if test "${nl_cv_build_optimized}" = "no"; then
+	CFLAGS="`echo ${CFLAGS} | sed -e 's,-O[[[:alnum:]]]*,-O0,g'`"
```

Comment:
man, it really bothers me that CFLAGS isn't an array.  this looks brittle.

consider a package named "Foo-Off".  we'd have trouble pulling headers from there, I think.

Reply:
Indeed.